File: py/entity/get_location_by_zipcode_entity.py
# Ziptastic SDK GetLocationByZipcode entity

from utility.voxgig_struct import voxgig_struct as vs
from core import helpers


class GetLocationByZipcodeEntity:

    def __init__(self, client, entopts=None):
        if entopts is None:
            entopts = {}
        if "active" not in entopts:
            entopts["active"] = True
        elif entopts["active"] is False:
            pass  # keep false
        else:
            entopts["active"] = True

        self._name = "get_location_by_zipcode"
        self._client = client
        self._utility = client.get_utility()
        self._entopts = entopts
        self._data = {}
        self._match = {}

        self._entctx = self._utility.make_context({
            "entity": self,
            "entopts": entopts,
        }, client.get_root_ctx())

        self._utility.feature_hook(self._entctx, "PostConstructEntity")

    def get_name(self):
        return self._name

    def make(self):
        opts = {}
        for k, v in self._entopts.items():
            opts[k] = v
        return GetLocationByZipcodeEntity(self._client, opts)

    def data_set(self, args=None):
        if args is not None:
            self._data = helpers.to_map(vs.clone(args)) or {}
            self._utility.feature_hook(self._entctx, "SetData")

    def data_get(self):
        self._utility.feature_hook(self._entctx, "GetData")
        return vs.clone(self._data)

    def match_set(self, args=None):
        if args is not None:
            self._match = helpers.to_map(vs.clone(args)) or {}
            self._utility.feature_hook(self._entctx, "SetMatch")

    def match_get(self):
        self._utility.feature_hook(self._entctx, "GetMatch")
        return vs.clone(self._match)

    
    def load(self, reqmatch, ctrl=None):
        utility = self._utility
        ctx = utility.make_context({
            "opname": "load",
            "ctrl": ctrl,
            "match": self._match,
            "data": self._data,
            "reqmatch": reqmatch,
        }, self._entctx)

        def post_done():
            if ctx.result is not None:
                if ctx.result.resmatch is not None:
                    self._match = ctx.result.resmatch
                if ctx.result.resdata is not None:
                    self._data = helpers.to_map(vs.clone(ctx.result.resdata)) or {}

        return self._run_op(ctx, post_done)



    

    

    

    

    def _run_op(self, ctx, post_done):
        utility = self._utility

        # #PrePoint-Hook

        point, err = utility.make_point(ctx)
        ctx.out["point"] = point
        if err is not None:
            return utility.make_error(ctx, err)

        # #PreSpec-Hook

        spec, err = utility.make_spec(ctx)
        ctx.out["spec"] = spec
        if err is not None:
            return utility.make_error(ctx, err)

        # #PreRequest-Hook

        resp, err = utility.make_request(ctx)
        ctx.out["request"] = resp
        if err is not None:
            return utility.make_error(ctx, err)

        # #PreResponse-Hook

        resp2, err = utility.make_response(ctx)
        ctx.out["response"] = resp2
        if err is not None:
            return utility.make_error(ctx, err)

        # #PreResult-Hook

        result, err = utility.make_result(ctx)
        ctx.out["result"] = result
        if err is not None:
            return utility.make_error(ctx, err)

        # #PreDone-Hook

        post_done()

        return utility.done(ctx)

File: rb/entity/get_location_by_zipcode_entity.rb
# Ziptastic SDK GetLocationByZipcode entity

require_relative '../utility/struct/voxgig_struct'
require_relative '../core/helpers'

class GetLocationByZipcodeEntity
  def initialize(client, entopts = nil)
    entopts ||= {}
    if entopts["active"].nil?
      entopts["active"] = true
    elsif entopts["active"] == false
      # keep false
    else
      entopts["active"] = true
    end

    @_name = "get_location_by_zipcode"
    @_client = client
    @_utility = client.get_utility
    @_entopts = entopts
    @_data = {}
    @_match = {}

    @_entctx = @_utility.make_context.call({
      "entity" => self,
      "entopts" => entopts,
    }, client.get_root_ctx)

    @_utility.feature_hook.call(@_entctx, "PostConstructEntity")
  end

  def get_name
    @_name
  end

  def make
    opts = @_entopts.dup
    GetLocationByZipcodeEntity.new(@_client, opts)
  end

  def data_set(args)
    if args
      @_data = ZiptasticHelpers.to_map(VoxgigStruct.clone(args)) || {}
      @_utility.feature_hook.call(@_entctx, "SetData")
    end
  end

  # @return [GetLocationByZipcode, Hash] the current GetLocationByZipcode data
  def data_get
    @_utility.feature_hook.call(@_entctx, "GetData")
    VoxgigStruct.clone(@_data)
  end

  def match_set(args)
    if args
      @_match = ZiptasticHelpers.to_map(VoxgigStruct.clone(args)) || {}
      @_utility.feature_hook.call(@_entctx, "SetMatch")
    end
  end

  # @return [Hash] the current match filter (any subset of GetLocationByZipcode fields)
  def match_get
    @_utility.feature_hook.call(@_entctx, "GetMatch")
    VoxgigStruct.clone(@_match)
  end

  
  # Load a single GetLocationByZipcode.
  #
  # @param reqmatch [GetLocationByZipcodeLoadMatch, Hash, nil] match criteria (id/query fields);
  #   optional — an entity with no id-like key loads with no match (nil is treated
  #   as an empty match, so client.GetLocationByZipcode.load works with no args).
  # @param ctrl [Object, nil] optional per-call control
  # @return [GetLocationByZipcode, Hash] the loaded GetLocationByZipcode; raises ZiptasticError on failure
  def load(reqmatch = nil, ctrl = nil)
    utility = @_utility
    ctx = utility.make_context.call({
      "opname" => "load",
      "ctrl" => ctrl,
      "match" => @_match,
      "data" => @_data,
      "reqmatch" => reqmatch,
    }, @_entctx)

    _run_op(ctx) do
      if ctx.result
        @_match = ctx.result.resmatch if ctx.result.resmatch
        if ctx.result.resdata
          @_data = ZiptasticHelpers.to_map(VoxgigStruct.clone(ctx.result.resdata)) || {}
        end
      end
    end
  end



  

  

  

  

  private

  def _run_op(ctx, &post_done)
    utility = @_utility

    # #PrePoint-Hook

    point, err = utility.make_point.call(ctx)
    ctx.out["point"] = point
    return utility.make_error.call(ctx, err) if err

    # #PreSpec-Hook

    spec, err = utility.make_spec.call(ctx)
    ctx.out["spec"] = spec
    return utility.make_error.call(ctx, err) if err

    # #PreRequest-Hook

    resp, err = utility.make_request.call(ctx)
    ctx.out["request"] = resp
    return utility.make_error.call(ctx, err) if err

    # #PreResponse-Hook

    resp2, err = utility.make_response.call(ctx)
    ctx.out["response"] = resp2
    return utility.make_error.call(ctx, err) if err

    # #PreResult-Hook

    result, err = utility.make_result.call(ctx)
    ctx.out["result"] = result
    return utility.make_error.call(ctx, err) if err

    # #PreDone-Hook

    post_done.call

    utility.done.call(ctx)
  end
end
